src/engine/MaxConstraint.h

```diff
@@ -190,54 +204,24 @@ class MaxConstraint : public PiecewiseLinearConstraint
     */
     String serializeToString() const override;
 
-    /*
-     * Returns a boolean value indicating if at least one input variable was eliminated (True)
-     * or not (False)
-     */
-    bool wereVariablesEliminated() const;
-
-
     bool isImplication() const override;
 
 private:
     unsigned _f;
     Set<unsigned> _elements;
     Set<unsigned> _initialElements;
 
-    double _maxLowerBound;
-    bool _obsolete;
-    bool _eliminatedVariables;
-    double _maxValueOfEliminated;
-
-    /*
-       Functions that abstract away _maxIndex and _maxIndexSet, and use the
-       refactored PhaseStatus to store this information.
-
-       Assumes that the total number of variables does not reach the value of
-       MAX_PHASE_ELIMINATED.
-     */
-    inline bool maxIndexSet() const
-    {
-        return getPhaseStatus() != PHASE_NOT_FIXED;
-    }
-
-    inline void setMaxIndex( unsigned variable )
-    {
-        setPhaseStatus( variableToPhase( variable ) );
-    }
+    Map<unsigned, unsigned> _auxToElement;
+    Map<unsigned, unsigned> _elementToAux;
 
-    inline unsigned getMaxIndex() const
-    {
-        ASSERT( maxIndexSet() );
-        return phaseToVariable( getPhaseStatus() );
-    }
+    bool _obsolete;
 
-    inline void clearMaxIndex()
-    {
-        setPhaseStatus( PHASE_NOT_FIXED );
-    }
+    // We eagerly keep track of the max lower bound of the inputs.
+    double _maxLowerBound;
 
-    void resetMaxIndex();
+    // We keep track of eliminated variables and the maximal value.
+    bool _haveFeasibleEliminatedPhases;
```

Comment:
Adjectives feasible and eliminated might sounds a bit contradictory. Can we come up with a simpler name or just elaborate on the exact meaning in the comments.

Reply:
Added more documentation. I thought more about "_isEliminatedPhaseFeasible" but I didn't like it too much either because it doesn't clearly signal that the flag is on iff 1) there are eliminated phases; and 2) one of them is feasible.